cli/global_state.rs

```diff
@@ -113,54 +120,91 @@ impl GlobalState {
   ) -> Result<(), AnyError> {
     let module_specifier = module_specifier.clone();
 
-    let mut module_graph_loader = ModuleGraphLoader::new(
-      self.file_fetcher.clone(),
-      maybe_import_map,
-      permissions.clone(),
-      is_dyn_import,
-      false,
-    );
-    module_graph_loader
-      .add_to_graph(&module_specifier, maybe_referrer)
-      .await?;
-    let module_graph = module_graph_loader.get_graph();
-
-    let out = self
-      .file_fetcher
-      .fetch_cached_source_file(&module_specifier, permissions.clone())
-      .expect("Source file not found");
-
-    let module_graph_files = module_graph.values().collect::<Vec<_>>();
-    // Check integrity of every file in module graph
-    if let Some(ref lockfile) = self.lockfile {
-      let mut g = lockfile.lock().unwrap();
+    if self.flags.no_check {
+      debug!("Transpiling root: {}", module_specifier);
+      let handler =
+        Rc::new(RefCell::new(FetchHandler::new(&self.flags, &permissions)?));
+      let mut builder = GraphBuilder::new(handler, maybe_import_map);
+      builder.insert(&module_specifier).await?;
+      let mut graph = builder.get_graph();
+
+      // TODO(kitsonk) this needs to move, but CompilerConfig is way too
+      // complicated to use here.
+      let maybe_config = if let Some(path) = self.flags.config_path.clone() {
+        let cwd = std::env::current_dir()?;
+        let config_file = cwd.join(path);
+        let config_path = config_file.canonicalize().map_err(|_| {
+          io::Error::new(
+            io::ErrorKind::InvalidInput,
+            format!(
+              "Could not find the config file: {}",
+              config_file.to_string_lossy()
+            ),
+          )
+        })?;
+        let config_str = fs::read_to_string(config_path)?;
+
+        Some(config_str)
+      } else {
+        None
+      };
```

Comment:
Hopefully this is temporary

Reply:
Yeah, it will get refactored out to something when there is a second consumer, or I will simplify compiler config.